vllm/multimodal/processing.py

```diff
@@ -205,10 +230,54 @@ def mode(self) -> UpdateMode:
         """Defines how to update the prompt."""
         raise NotImplementedError
 
-    def bind(self, tokenizer: AnyTokenizer) -> "BoundPromptUpdate":
-        return BoundPromptUpdate(
-            _origin=self,
-            tokenizer=tokenizer,
+    def _resolve_target(
+        self,
+        tokenizer: AnyTokenizer,
+        item_idx: int,
+    ) -> Union["_BoundPromptSequence", PromptIndex]:
+        target = self.target
+        if callable(target):
+            target = target(item_idx)
+
```

Comment:
```suggestion
        if callable(target):
            target = target(item_idx)
        elif isinstance(target, list):
            target = target[item_idx]
```
If we allow target to be a list, I think we can avoid `target=image_tokens.__getitem__` and use `target=image_tokens` in phi-4-mm.

Reply:
That's true, but I want to be more explicit about how the dynamic resolution is achieved.